chapters/glossary.md

```diff
@@ -59,6 +59,10 @@ Sometimes when people use the term "network," they mean "internetwork." Since "i
 
 A router is any machine whose purpose is to connect networks together. It does so by forwarding packets further toward their destination. Each router has a routing table which serves much like a sign post on a highway: it tells the router where to send packets next on their way to their final destination. Each router makes decisions on its own for the most efficient way to send the packet to its destination. The internet, as we know today, is not possible without numerous routers facilitating the requests.
 
+## routing table
+
+A routing table is a list of known network IP address ranges and potentially a default gateway that lives on every machine connected to an internet. The routing table is used to know how to send packets across an internet from a source machine to a destination machine. The routing table shows network prefixes and subnets and associates them with a next hop that can forward packets towards the destination. If the routing table does not have a matching route for the destination IP address, the request can be sent out the default gateway in the hopes that another machine on that internet will know how to get the packets to their destination.
+
```

Comment:
```suggestion
A routing table is a list of known network IP address ranges and potentially a default gateway that lives on every machine connected to an internet.  The routing table shows network prefixes and subnets and associates them with a next hop that can forward packets towards the destination. If the routing table does not have a matching route for the destination IP address, the request can be sent out the default gateway in the hopes that another machine on that internet will know how to get the packets to their destination.
```

- I guess it's okay to distinguish between two types of routes: "IP address ranges" and a "default gateway". But really they are the same thing because "a default gateway" is nothing more than a route to "0.0.0.0/0" which is also an IP address range. 
- I didn't love "The routing table is used to know how to send packets across an internet from a source machine to a destination machine." That strikes me as a little too strong of a statement, since all it does is tell you "where to go next." You could say, "the set of all routing tables across the internet is used to send packets across an internet from a source machine to a destination machine." if you like.
- Also the statement, "If the routing table does not have a matching route for the destination IP address, the request can be sent out the default gateway in the hopes that another machine on that internet will know how to get the packets to their destination." is *okay*, but remember the "default gateway" *is* a matching route.

This all makes me think that we should probably do a chapter on network summarization next!

Reply:
you're totally right on all of that. Here's what I have now, but I don't like it. It feels too stodgy. If I didn't know what a routing table was, this would only make me more confused: 

```
A routing table is a list of known network IP address ranges, and every machine connected to an internet has a routing table to reference. Machines use routing tables to determine what the next "hop" should be when they're sending packets towards a destination IP address. If the routing table does not have a matching route for the destination IP address, the request can be sent out the default gateway in the hopes that another machine on that internet will know how to get the packets to their destination. Check [prefixes-and-subnet-masks.md in the appendix](./appendix/prefixes-and-subnet-masks.md) for more details on how routing tables work.
```

thoughts?

Reply:
Not bad! Here's my attempt. Not sure if it's any better or not:

A routing table is a list of networks and where-to-go-next for each of those networks. It just says the equivalent of, "To go to Albuquerque, take a left; to go to LA or San Francisco, go straight." Machines look at packets they're sending and use routing tables to determine the next turn they should take toward that packet's destination (or "hop" in network-speak).

Every machine connected to an internet has one of these routing tables to reference, even if the routing table just says, "For every network you want to go to, always use the one router that is on your network."  Check [prefixes-and-subnet-masks.md in the appendix](https://github.com/psbanka/build-your-own-internet/pull/appendix/prefixes-and-subnet-masks.md) for more details on how routing tables work.